### PR TITLE
Strict mode for JSON

### PR DESCRIPTION
**Pull request by maxtaco**
Require only ASCII characters, and to have sorted dictionary keys,
and also to have no repeated keys.

In service to keybase/keybase#428

Seems to work with a few tweaks to field ordering in the Go client.

Blocked on keybase/keybase#320 for further testing

### CODE REVIEW

**Comment by maxtaco** [comment]
Can you review @Sidnicious?

File: src/base.iced [hunk 1]
```diff
@@ -54,7 +54,7 @@ exports.cieq = cieq = (a,b) -> (a? and b? and (a.toLowerCase() is b.toLowerCase(
 
 class Verifier
 
-  constructor : ({@armored, @id, @short_id, @skip_ids, @make_ids}, @sig_eng, @base) ->
+  constructor : ({@armored, @id, @short_id, @skip_ids, @make_ids, @strict}, @sig_eng, @base) ->
```
Comment (Sidnicious) [inline src/base.iced]: It doesn't look like this argument does anything.

File: src/base.iced [hunk 2]
```diff
@@ -112,14 +112,28 @@ class Verifier
 
   #---------------
 
+  _check_json_formatting : (cb) ->
```
Comment (Sidnicious) [inline src/base.iced]: …and this doesn't have any callers. Did I miss something?

File: src/base.iced [hunk 3]
```diff
@@ -112,14 +112,28 @@ class Verifier
...
-    err = new Error "Couldn't parse JSON signed message: #{e.message}" if e?
-    if not err?
-      jsons = jsons.toString('utf8')
-      await @base._v_check {@json}, defer err
-    cb err, @json, jsons
+    json_str_buf = @payload
+    json_str_utf8 = json_str_buf.toString('utf8')
+    err = null
+    if not /^[\x20-\x7e]+$/.test json_str_utf8
+      err = new Error "All JSON proof characters must be in the ASCII set"
```
Comment (Sidnicious) [inline src/base.iced]: Maybe this error should mention that non-ASCII characters can be included if escaped?

File: src/base.iced (hunk 2 above)
Comment (maxtaco) [inline src/base.iced]: Nope, just dead code...

File: src/base.iced (hunk 1 above)
Comment (maxtaco) [inline src/base.iced]: The check below should only happen in strict mode...

**Pull request merged**